addons/binding/org.openhab.binding.lgwebos/src/main/java/org/openhab/binding/lgwebos/internal/discovery/LGWebOSDiscovery.java

```diff
@@ -161,16 +165,55 @@ public String getPackageName() {
 
     @Override
     public InetAddress getIpAddress() {
+        if (localInetAddressesOverride.isPresent()) {
+            return localInetAddressesOverride.get();
+        }
+
+        Optional<InetAddress> inetAddressFromNetworkAddressService = getInetAddressFromNetworkAddressService();
```

Comment:
These lines make it quit heavy to use an optional here (also optionals are not common in the openHAB / ESH codebase the project opted for null / non-null annotations instead)

In the end you used 4 extra lines to write

```java
return getInetAddressFromNetworkAddressService().orElse(null); 
```

And the old implementation had the `orElse(null)` 'build in', so maybe the fallback can just keep returning null given you are actually not interested in the optional.

Reply:
Well, I actually quite like the optional feature, and since this is not part of any public interface I was hoping it is ok. I condensed getIpAddress() down to one line.